Developer docs for Telephony inline hook (#3019)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/telephony-hook/index.md

---
title: Telephony Inline Hook Reference
excerpt: Customizes Okta's flows that send SMS or Voice messages
---

> 

Reviewer: ```suggestion
<ApiLifecycle access="beta" />

```

> **Note:** Administrators can use the [Okta System Log](/docs/reference/api/system-log/) to view errors. See the [Troubleshooting](/docs/concepts/inline-hooks/#troubleshooting) section in the Inline Hooks concept piece for more information on the events related to Inline Hooks that the Okta System Log captures.

- When there is a communication failure with the external service, a timeout for example, the inline hook operation is skipped, and the OTP is delivered to the requester using Okta's telephony provider(s).

  **Who can see this error?** Administrators

- When the external service returns a response with any other HTTP status code besides `200`, the inline hook operation is skipped, and the OTP is delivered to the requester using Okta's telephony provider(s).

  **Who can see this error?** Administrators

- When the external service returns an error object in the response, the telephony inline hook flow fails with no OTP delivered using the hook.

  **Who can see this error?** Administrators, developers, and end users

> - When a hook response is malformed or couldn't be mappeed to the expected API response, the inline hook operation is skipped.

Reviewer: ```suggestion
- When a hook response is malformed or couldn't be mapped to the expected API response, the inline hook operation is skipped.
```

- JSON objects that are contained in the outbound request from Okta to your external service

- JSON objects that you can include in your response

This information is specific to the Telephony Inline Hook, one type of Inline Hook supported by Okta.

## See also

For a general introduction to Okta Inline Hooks, see [Inline Hooks](/docs/concepts/inline-hooks/).

For information on the API for registering external service endpoints with Okta, see [Inline Hooks Management API](/docs/reference/api/inline-hooks/).

For steps to enable this Inline Hook, see below, [Enabling a Token Inline Hook](#enabling-a-token-inline-hook).

## About

The Okta Telephony Inline Hook allows you to integrate your own custom code into several of Okta's flows that send SMS or Voice (CALL) messages.
You can integrate this hook with Enrollment, Authentication, and Recovery flows that involve the Phone
authenticator. While the One-Time Passcode (OTP) is sent to the requester, Okta calls your external
service to deliver the OTP, and your service can respond with commands that indicate success or failure in delivering the OTP.

Only one active telephony inline hook is allowed to be created for any org.

> All telephony hooks require authentication field and secret to be specified while creating the hook. It is passed as the

Reviewer 2: ```suggestion
When you create a Telephony Inline Hooks, you must include the `authScheme` parameter. See [Create Inline Hook](/docs/reference/api/inline-hooks/#create-inline-hook) and the [authScheme object](/docs/reference/api/inline-hooks/#authscheme-object) for more information.
```

---
title: Telephony Inline Hook Reference
excerpt: Customizes Okta's flows that send SMS or Voice messages.
---

# Telephony Inline Hook Reference

This page provides reference documentation for:

- JSON objects that are contained in the outbound request from Okta to your external service

- JSON objects that you can include in your response

This information is specific to the Telephony Inline Hook, one type of Inline Hook supported by Okta.

## See also

For a general introduction to Okta Inline Hooks, see [Inline Hooks](/docs/concepts/inline-hooks/).

For information on the API for registering external service endpoints with Okta, see [Inline Hooks Management API](/docs/reference/api/inline-hooks/).

> For steps to enable this Inline Hook, see below, [Enabling a Token Inline Hook](#enabling-a-token-inline-hook).

Author: Yes not applicable since for telephony its enabled/ACTIVE by default.

          "transactionMetadata":"Duration=300ms"
        }
      ]
    }
  ]
}
```

### Sample response for failed OTP delivery

```json
{
  "error":{
    "errorSummary":"Failed to deliver SMS OTP to test.user@okta.com",
    "errorCauses":[
      {
        "errorSummary":"Provider could not deliver OTP",
        "reason":"The content of the message is not supported",
        "location":"South Africa"
      }
    ]
  }
}
```

## Timeout behavior

After receiving the Okta request, if there is a response timeout, the Okta process flow proceeds with trying to send the
OTP using Okta's telephony provider(s). See [Troubleshooting](#troubleshooting).

## Troubleshooting

This section covers what happens when a telephony inline hook flow fails either due to the external inline hook service returning an error object or not returning a successful response.

> **Note:** Administrators can use the [Okta System Log](/docs/reference/api/system-log/) to view errors. See the [Troubleshooting](/docs/concepts/inline-hooks/#troubleshooting) section in the Inline Hooks concept piece for more information on the events related to Inline Hooks that the Okta System Log captures.

> - When there is a communication failure with the external service, a timeout for example, the inline hook operation is skipped.

Reviewer 2: ```suggestion
- When there is a communication failure with the external service, such as a timeout, the inline hook operation is skipped, and the OTP is delivered to the requester using Okta's telephony provider(s).
```

>   OTP is delivered to requester using Okta's telephony provider(s).

Reviewer 2: ```suggestion

```

  **Who can see this error?** Administrators

> - When the external service returns a response with any other HTTP status code besides `200`, the inline hook operation is skipped.

Reviewer 2: ```suggestion
- When the external service returns a response with any other HTTP status code besides `200`, the inline hook operation is skipped, and the OTP is delivered to the request using Okta's telephony provider(s).
```

  OTP is delivered to requester using Okta's telephony provider(s).

  **Who can see this error?** Administrators

> - When the external service returns an error object in the response, the telephony inline hook flow fails with no OTP delivered using the hook.

Reviewer 2: ```suggestion
- When the external service returns an error object in the response, the telephony inline hook flow fails with no OTP delivered.
```

  **Who can see this error?** Administrators, developers, and end users.

> - When a hook response is malformed or could not be mappeed to expected API response, the inline hook operation is skipped.

Reviewer 2: ```suggestion
- When a hook response is malformed or couldn't be mapped to the expected API response, the inline hook operation is skipped, and the OTP is delivered to the requester using Okta's telephony provider(s). There may be scenarios where the requester may receive duplicate SMS/CALL messages because of the external service and Okta provider(s) both attempting to deliver the OTP.
```

>   The OTP is delivered to requester using Okta's telephony provider(s). There may be scenarios where requester may receive

Reviewer 2: ```suggestion

```

>   duplicate SMS/CALL because of external service and Okta provider(s) both attempting to deliver the OTP.

Reviewer 2: ```suggestion

```

  **Who can see this error?** Administrators

  The following actions result in an error:

>   - Using an invalid status in response

Reviewer 2: ```suggestion
  - Using an invalid status in the response
```

  - Attempting to add an active telephony hook when one already exists

>   - Not including auth scheme in request header

Reviewer 2: ```suggestion
  - Not including the `authScheme` in the request header
```
